### DISCUSSION TITLE
v1.113.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#12151] [Announcements]
# v1.113.0

> [!WARNING]
> ## Breaking changes
> For **OAuth users**, please replace `app.immich:/` with `app.immich:///oauth-callback` for the Redirect URI in your OAuth provider settings

## Highlights

Welcome to release `v1.113.0` of Immich! This is one of the biggest releases yet, introducing some of the most requested features since the early days of Immich. Let's dive right into what we have in place for this release:

- Folder view
- Tags
- Timeline improvements
- Library refresh stability
- Mobile album sync

### Folder view

You can now browse your photos and videos by folder like in a file explorer. You can use the storage template migration feature for the best experience with uploaded assets in this view.

This feature is especially useful for scanned photos, which are difficult to put in a timeline. It has been a long-requested feature.

You can enable this feature from the `Users Settings > Features > Folders`. 

![folder-enabled](https://github.com/user-attachments/assets/aab4a16a-0888-49de-b51a-785ef557261a)

The UI is currently only available for the web; mobile will come in a subsequent release.

![folder](https://github.com/user-attachments/assets/2786da30-83b3-4168-bb98-180dad53a6a4)

![folder-2](https://github.com/user-attachments/assets/163a3ee8-3c3d-4c98-abb6-9c94eb4b362c)

### Tags

Immich now supports hierarchical tags, with the ability to read existing tags from the `TagList` and `Keywords` exif properties. Any changes to tags made through Immich are also written back to a sidecar file. You can re-run the metadata extraction jobs for all assets to import your existing tags.

You can enable this feature from the `Users Settings > Features > Tags`. 

![tag-enabled](https://github.com/user-attachments/assets/20a7a99e-13f6-46f2-8744-20dd95d8456b)

The UI is currently only available for the web; mobile will come in a subsequent release.

https://github.com/user-attachments/assets/d543f531-4b0e-4dcf-8918-e76c7f1b288b

### Timeline improvements

This release introduces a rewrite of the web timeline component. It can now handle a large number of assets in a single day or month and has been successfully tested with a very large data set (over a million assets). Photographers frequently request this since they can easily take thousands of photos in a given day.

With these performance improvements, you'll see fewer placeholders while loading, which will make for a more fluid scrolling and scrubbing experience.

### Library refresh stability

In relation to the previous point, the stability of library scanning has improved. Previously, you could run out of memory when starting a refresh with libraries containing millions of assets. Now, we queue the refresh jobs in batches. These enhancements won't make scanning go any faster, but they greatly reduce the likelihood of out-of-memory errors that would cause Immich to crash.

### Mobile album sync

You can now sync or mirror an album from your phone to the Immich server on your account. For example, if you select `Recents`, `Camera` and `Videos` album for backup, the corresponding album with the same name will be created on the server. Once the assets from those albums are uploaded, they will be put into the target albums automatically.

You can enable this feature from the album selection in the backup screen. 

<img src="https://github.com/user-attachments/assets/67b329f5-15ff-4128-af52-4a50df852a07" width="300" alt="sync-album"/>

For existing installations, you can sync the already uploaded assets by going to the backup screen and pressing the `Sync` button.

<img src="https://github.com/user-attachments/assets/ca0847fd-edab-4ebe-b8e1-093ce7e3cc37" width="300" alt="sync-button"/>

Have a wonderful weekend,

Cheers!
- - - -

## Support Immich

<p align="center">
<img src="https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbjY2eWc5Y2F0ZW56MmR4aWE0dDhzZXlidXRmYWZyajl1bWZidXZpcyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/87CKDqErVfMqY/giphy.gif" width="450" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich by purchasing a product key at [https://buy.immich.app](https://buy.immich.app).

Cheers! 🍻

### 🚨 Breaking Changes
* feat(server): granular permissions for api keys by @jrasm91 in https://github.com/immich-app/immich/pull/11824
* refactor(server): stacks by @jrasm91 in https://github.com/immich-app/immich/pull/11453
* fix(server): album statistics endpoint by @jrasm91 in https://github.com/immich-app/immich/pull/11924
* fix: remove `asset.resized` by @jrasm91 in https://github.com/immich-app/immich/pull/11983
* fix(mobile): use a valid OAuth callback URL by @qrkourier in https://github.com/immich-app/immich/pull/10832
### 🚀 Features
* feat: folder view by @davidakerr in https://github.com/immich-app/immich/pull/11880
* feat(web): Scroll to asset in gridview; increase gridview perf; reduce memory; scrollbar ticks in fixed position by @midzelis in https://github.com/immich-app/immich/pull/10646
* feat: loading screen, initSDK on bootstrap, fix FOUC for theme by @midzelis in https://github.com/immich-app/immich/pull/10350
* feat(mobile): preserve mobile album info on upload by @alextran1502 in https://github.com/immich-app/immich/pull/11965
* feat: tags by @jrasm91 in https://github.com/immich-app/immich/pull/11980
* feat(web): jump to timeline by @alextran1502 in https://github.com/immich-app/immich/pull/12117
### 🌟 Enhancements
* feat(server): do not automatically download android motion videos by @jrasm91 in https://github.com/immich-app/immich/pull/11774
* feat(web): pasting coordinates by @michelheusschen in https://github.com/immich-app/immich/pull/11866
* feat(web): drag and drop or paste directories for upload by @simkli in https://github.com/immich-app/immich/pull/11879
* feat(web): Left hand navigation for memories by @carlesalbasboix in https://github.com/immich-app/immich/pull/11913
* feat(web): my immich shortcut by @danieldietzler in https://github.com/immich-app/immich/pull/12007
* fix(web): show a clearer confirmation message when deleting an unnamed album by @Snowknight26 in https://github.com/immich-app/immich/pull/11988
* feat(format): nrw format by @avsm in https://github.com/immich-app/immich/pull/12048
* feat(web): restore scroll position on navigating back to search page by @alextran1502 in https://github.com/immich-app/immich/pull/12042
* feat(server): Storage template support album condition by @feyst in https://github.com/immich-app/immich/pull/12000
* fix(mobile): Changes in the UI for the image editor pages by @Yuvi-raj-P in https://github.com/immich-app/immich/pull/12018
* feat(web): announce notifications to screen readers by @ben-basten in https://github.com/immich-app/immich/pull/12071
* fix(server): don't crash when refreshing large libraries by @etnoy in https://github.com/immich-app/immich/pull/7934
* feat(server): sort images in duplicate groups by date by @GeoffreyFrogeye in https://github.com/immich-app/immich/pull/12094
* feat(ml): support dynamic scaling by @rkojedzinszky in https://github.com/immich-app/immich/pull/12065
* feat(web): navigate assets with gestures (next/prev) by @kaziu687 in https://github.com/immich-app/immich/pull/11888
* fix(mobile): allow create empty non-shared albums, add proper button colors by @src52 in https://github.com/immich-app/immich/pull/12103
* feat: user's features preferences by @alextran1502 in https://github.com/immich-app/immich/pull/12099
* chore(server): log path when generating external thumbnail by @etnoy in https://github.com/immich-app/immich/pull/12107
### 🐛 Bug fixes
* fix(web): focus trap inside portal by @michelheusschen in https://github.com/immich-app/immich/pull/11797
* fix(mobile): show correct notification icon for android by @karthikraja001 in https://github.com/immich-app/immich/pull/11863
* fix(web): show camera make in search options after searching by @michelheusschen in https://github.com/immich-app/immich/pull/11884
* fix(web): correctly populate the camera model search dropdown by @Snowknight26 in https://github.com/immich-app/immich/pull/11883
* fix(server): create shared album from the mobile app does not trigger send email invite by @alextran1502 in https://github.com/immich-app/immich/pull/11911
* fix(server): do not match live photos across libraries by @jrasm91 in https://github.com/immich-app/immich/pull/11952
* fix(web): rating stars accessibility by @ben-basten in https://github.com/immich-app/immich/pull/11966
* fix(mobile): Fix for incorrectly naming edited files and structure change by @Yuvi-raj-P in https://github.com/immich-app/immich/pull/11741
* fix: align camera model drop down behavior with other drop downs on web and mobile by @x24git in https://github.com/immich-app/immich/pull/11951
* fix(web): announce current theme to screen reader users by @ben-basten in https://github.com/immich-app/immich/pull/12039
* fix(web): show supporter badge for account less than 14 days by @alextran1502 in https://github.com/immich-app/immich/pull/12058
* fix(web): shared link expiration date accessibility by @ben-basten in https://github.com/immich-app/immich/pull/12060
* chore(web): ignore shortcut toggle when entering email and password by @alextran1502 in https://github.com/immich-app/immich/pull/12082
* chore(web): ensure goto is awaited for login page by @alextran1502 in https://github.com/immich-app/immich/pull/12087
* fix(server): ensure new exclusion patterns work by @etnoy in https://github.com/immich-app/immich/pull/12102
* fix(server): skip smtp validation if unchanged by @michelheusschen in https://github.com/immich-app/immich/pull/12111
* fix(mobile): long waiting time for login request when server is unreachable by @alextran1502 in https://github.com/immich-app/immich/pull/12100
* fix: user specific fields in asset search by @jrasm91 in https://github.com/immich-app/immich/pull/12125
* fix(web): Device list shows Ubuntu as unknown OS by @spfncer in https://github.com/immich-app/immich/pull/12127
* fix(web): reset asset grid after changing album order by @michelheusschen in https://github.com/immich-app/immich/pull/12139
### 📚 Documentation
* fix(docs): read-only affects XMP writing by @C-Otto in https://github.com/immich-app/immich/pull/11823
* docs: clarify external domain setting by @pikapower9080 in https://github.com/immich-app/immich/pull/11958
* docs: add Immich Kiosk and Immich Power Tools to Community Projects by @Tyree in https://github.com/immich-app/immich/pull/12055
* docs: mTLS/self signed FAQ entry by @mmomjian in https://github.com/immich-app/immich/pull/12074
* docs: external library deletion/edits by @mmomjian in https://github.com/immich-app/immich/pull/12079
* docs: sql query for duplicate files by @mmomjian in https://github.com/immich-app/immich/pull/12086
* docs: Documentation updates by @aviv926 in https://github.com/immich-app/immich/pull/11516
* docs: update roadmap by @jrasm91 in https://github.com/immich-app/immich/pull/12126
* fix: README_zh_CN.md link by @ttzytt in https://github.com/immich-app/immich/pull/12124
* docs(guide): nginx caching proxy by @pcouy in https://github.com/immich-app/immich/pull/12140

## New Contributors
* @aaronjrodrigues made their first contribution in https://github.com/immich-app/immich/pull/11851
* @karthikraja001 made their first contribution in https://github.com/immich-app/immich/pull/11863
* @simkli made their first contribution in https://github.com/immich-app/immich/pull/11879
* @carlesalbasboix made their first contribution in https://github.com/immich-app/immich/pull/11907
* @pikapower9080 made their first contribution in https://github.com/immich-app/immich/pull/11958
* @davidakerr made their first contribution in https://github.com/immich-app/immich/pull/11880
* @x24git made their first contribution in https://github.com/immich-app/immich/pull/11951
* @Tonux599 made their first contribution in https://github.com/immich-app/immich/pull/12027
* @avsm made their first contribution in https://github.com/immich-app/immich/pull/12048
* @Tyree made their first contribution in https://github.com/immich-app/immich/pull/12055
* @feyst made their first contribution in https://github.com/immich-app/immich/pull/12000
* @Tiefseetauchner made their first contribution in https://github.com/immich-app/immich/pull/11874
* @qrkourier made their first contribution in https://github.com/immich-app/immich/pull/10832
* @GeoffreyFrogeye made their first contribution in https://github.com/immich-app/immich/pull/12094
* @rkojedzinszky made their first contribution in https://github.com/immich-app/immich/pull/12065
* @kaziu687 made their first contribution in https://github.com/immich-app/immich/pull/11888
* @src52 made their first contribution in https://github.com/immich-app/immich/pull/12103
* @spfncer made their first contribution in https://github.com/immich-app/immich/pull/12127
* @ttzytt made their first contribution in https://github.com/immich-app/immich/pull/12124

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.112.1...

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.113.0'>v1.113.0</a>.</em>

**Comment by Tyree** [comment]
I finally contributed something! LOL! :-D
Immich is the best.

**Comment by deddc23efb** [comment]
How stable is this.  I've been holding off upgrading from 1.111 .

**Reply by nicomerl** [reply]
As always: Immich is not yet in stable mode so there might be bugs. With reading the release notes and upgrading i never had problems besides smaller Bugs since over a year.

**Reply by aamirazad** [reply]
I usually wait a few days before deploying so they can patch anything major

**Comment by Solarunit** [comment]
Folders is the best new feature. Would be nice to have option to exclude those folders with images without metadata from the timeline.
Also would be nice to have favorite folders where you can pin folders with actual images to avoid searching for the right folder in the tree of folders. Or manually create a folder with a path to root folder with images.

UPD: or option to display all images in all sub-folders when you are in root folder.

**Reply by PanSzelescik** [reply]
Sorting files and folders would be nice too

**Comment by Solarunit** [comment]
Tags is another cool feature. Maybe option to create tag on the fly? When you already have image open.

**Reply by rjgrandy** [reply]
Incredible update!

Also tagging a group of photos would be super useful, like if i search for something then select a bunch of the photos, apply the tag to all of the selected photos

**Reply by Vyerni** [reply]
100% agree on the tagging. I want to go through an assign tags to a lot of my images, to create some sort of organisation, but having to add a tag at a time to each photo is very painful.

It's an incredible feature though, thank you

**Reply by jrasm91** [reply]
Yeah, I wanted to implement this, but ran out of steam. I think next iterations of the feature would include creating one or many tags from that original modal and then adding them immediately to the selected asset(s).

**Reply by jrasm91** [reply]
You can multi-select and add tags on the main timeline already, unless I am missing something.

**Comment by MJHeijster** [comment]
Awesome! Glad I read the notes though due to the breaking change.

**Comment by Zindswini** [comment]
Incredible progress! After seeing that the mobile album sync was merged I immediately bought a key!

Thanks for the great work :heart:

**Comment by stibbzy** [comment]
I'm not seeing the Features or Folders options under User Settings after upgrading... am I missing something?

**Reply by alextran1502** [reply]
You probably haven't pulled the new image. What does the version on the lower left corner say?

**Reply by stibbzy** [reply]
Pretty sure I have.. it says “Server Online v1.113.0”

**Reply by alextran1502** [reply]
Did you go to the `Account Settings`? Perhaps you accessed the Admin Setting instead?

![image](https://github.com/user-attachments/assets/6a592087-d9ad-41ef-844a-54a6e54d6e82)

**Reply by stibbzy** [reply]
Ahh yep, that was it lol. Sorry about that!

**Comment by SHU-red** [comment]
I love the new tag section
Did not know that something like this was in work but it was a very imporant feature for me in GPhotos to add text to photos and find them (like  e.g. "ID")

Now in can structure them in tags which is damn awesome!!!

**Comment by chriswa81** [comment]
The synchronization of the photos in the album works very well. However, somehow only in one direction. When I add a photo to an album on my cell phone (iphone), it is synchronized correctly in immich. If I remove the photo from the album on the cell phone, the photo remains in the album in immich.

**Reply by bo0tzz** [reply]
That's working as expected.

**Reply by qSib** [reply]
That how it should work. If you remove it from the Immich app it should prompt to remove it from the device.

**Comment by jenda69** [comment]
Yay, folder view!

Out of curiosity. If I enable folder view, but don't enable it on sidebar, what does it do? Does it add link to the view anywhere or is it only accessible by manually opening `/folders`? (edit: Well, I can access that even if it's not enabled at all.)
It would be cool if it would show up in "Appears in" section of the Info panel just like albums do - that's the only reason I'll keep using the "albums from folders" script for now.

Will there be a way to select a "home directory"? Right now it's a bit annoying to go through "/usr/src/app/external/username/" until I get to my photos. (edit: Solved that by moving my external library to /photos/ folder :-) )

**Reply by C-Otto** [reply]
I think it would be nice to automatically select the first directory that contains an asset or more the one subdirectory.

**Reply by alextran1502** [reply]
The enabled for those feature are placed holder for future expansion of each feature where you can mass toggle sub features

**Reply by davidakerr** [reply]
I can add a feature to trim the left side of external library path to the last path element in the specified external path.

**Reply by petersuba** [reply]
This would be great - in my case on a NAS I have to always navigate to /usr/src/app/external/username before I get anything useful. thinking further, I think there are probably 2 different enhancement possibilities here:
- trimming the path to the first meaningful directory (could be a setting)
- defaulting the path when switching to the Folder view. Ideally it would be last displayed path (i.e. before navigating away to use e.g. timeline view) but could also be just a custom setting to what the default would be (i.e. at the moment this is simply / but if I can set it to /usr/src/app/external/username then even if the path takes up real estate, at least I don't need to navigate this path with 5 clicks every time I switch the view)

**Comment by SeanAshe** [comment]
Can't add tag for selected photo in filtered mode like albums and search

**Comment by muammar** [comment]
No Asset On Device - Abort Backup Process

This release broke the ability to backup on my iPhone.

**Reply by alextran1502** [reply]
Can you provide more context of the bug by opening the issue?

**Comment by acbarrentine** [comment]
When I turned on folders, it took me a little while to figure out why some folders showed up and some did not. Eventually I worked out that a folder would be hidden if every image within it (recursively) was Archived. You can type in the name of a folder (in the URL) that is hidden by this filter and it will show up.

Today I unarchived all the photos causing this quirk, but it turns out those folders are still hidden. Is there a way to "kick" the folder discovery mechanism to see if these previously hidden ones should now show?

**Reply by acbarrentine** [reply]
Oh. Answered my own question. If I do the URL hack, it made the previously hidden folders pop into visibility. Appears to stick, too.

**Reply by mertalev** [reply]
The folder sidebar gets cached for a bit, so it can take a hard refresh for it to show changes.
